frontend/src/components/layout/whatsnew/WhatsNewList.tsx

```diff
@@ -0,0 +1,111 @@
+import { useLocalization } from "@fluent/react";
+import { Key, ReactNode, useRef } from "react";
+import { useMenu, useMenuItem } from "react-aria";
+import { Item, TreeProps, TreeState, useTreeState } from "react-stately";
+import EmptyStateHero from "./images/empty-hero.png";
+import styles from "./WhatsNewList.module.scss";
+import { WhatsNewEntry } from "./WhatsNewMenu";
+
+export type Props = {
+  entries: WhatsNewEntry[];
+  onSelect: (entry: WhatsNewEntry) => void;
+  "aria-label": string;
+};
+
+export const WhatsNewList = (props: Props) => {
+  const { l10n } = useLocalization();
+
+  if (props.entries.length === 0) {
+    return (
+      <div className={styles["empty-message"]}>
+        <img src={EmptyStateHero.src} alt="" />
+        <p>{l10n.getString("whatsnew-empty-message")}</p>
+      </div>
+    );
+  }
+
+  return (
+    <WhatsNewListMenu
+      aria-label={props["aria-label"]}
+      onSelect={(key) => {
+        const selectedItem = props.entries.find(
+          (entry) => encodeURI(entry.title) === key
+        );
+        /* istanbul ignore if [Types can't represent that a `selectedItem` should always be found, but it should.] */
+        if (!selectedItem) {
+          // TODO: This shouldn't happen; can we log this somewhere if it does?
+          return;
+        }
+        props.onSelect(selectedItem);
+      }}
```

Comment:
Hmm yeah good point, just changed this to just tell TS that `selectedItem` will always be found: bd0e2b52

Reply:
Sorry I mis-understood this code on the first review. I thought ALL of the `onSelect` functions were written just to catch this problem. But I understand better now that the `onSelect` eventually calls `WhatsNewDashboard.selectEntry`, which calls `setExpandedEntry` and `entry.dismissal.dismiss()`.

So, rather than use the un-safe `selectedItem!`, can we go back to this but include a `console.error`?

```typescript
        if (!selectedItem) {
          console.error(`WhatsNewListMenu.onSelect: ${key} not found in WhatsNewEntry array.`);
          return;
        }
```

Reply:
c68280ed